tests/director/external-services-mock-integration/async_api_delete_test.go

```diff
@@ -0,0 +1,171 @@
+	appName := fmt.Sprintf("app-async-del-%s", time.Now().Format("060102150405"))
+	appInput := graphql.ApplicationRegisterInput{
+		Name:         appName,
+		ProviderName: ptr.String("compass"),
+		Webhooks: []*graphql.WebhookInput{
+			{
+				Type:           graphql.WebhookTypeUnregisterApplication,
+				Mode:           webhookModePtr(graphql.WebhookModeAsync),
+				URLTemplate:    str.Ptr(fmt.Sprintf("{ \\\"method\\\": \\\"GET\\\", \\\"path\\\": \\\"%s\\\" }", deleteFullPath)),
+				OutputTemplate: str.Ptr(fmt.Sprintf("{ \\\"location\\\": \\\"%s\\\", \\\"success_status_code\\\": 200, \\\"error\\\": \\\"{{.Body.error}}\\\" }", operationFullPath)),
+				StatusTemplate: str.Ptr("{ \\\"status\\\": \\\"{{.Body.status}}\\\", \\\"success_status_code\\\": 200, \\\"success_status_identifier\\\": \\\"SUCCEEDED\\\", \\\"in_progress_status_identifier\\\": \\\"IN_PROGRESS\\\", \\\"failed_status_identifier\\\": \\\"FAILED\\\", \\\"error\\\": \\\"{{.Body.error}}\\\" }"),
+			},
+		},
+	}
+
+	t.Log("Get Dex token")
+	dexToken, err := idtokenprovider.GetDexToken()
+	require.NoError(t, err)
+	dexGraphQLClient := gql.NewAuthorizedGraphQLClient(dexToken)
+
+	t.Log(fmt.Sprintf("Registering application: %s", appName))
+	appInputGQL, err := tc.Graphqlizer.ApplicationRegisterInputToGQL(appInput)
+	require.NoError(t, err)
+	registerRequest := fixRegisterApplicationRequest(appInputGQL)
+	app := graphql.ApplicationExt{}
+	err = tc.RunOperationWithCustomTenant(ctx, dexGraphQLClient, testConfig.DefaultTenant, registerRequest, &app)
+	require.NoError(t, err)
+
+	defer deleteApplicationOnExit(t, ctx, dexGraphQLClient, app.ID, testConfig.DefaultTenant)
+
+	t.Log("Unlock the mock application webhook")
+	unlockWebhook(t, operationFullPath)
+	isInDesiredState, webhookResponse := isWebhookOperationInDesiredState(t, operationFullPath, webhook.OperationResponseStatusOK)
+	require.True(t, isInDesiredState, fmt.Sprintf("Webhook responded with: %s. Expected state: %s", webhookResponse, webhook.OperationResponseStatusOK))
+
+	t.Log("Start async Delete of application")
+	unregisterAsyncApplicationInTenant(t, ctx, dexGraphQLClient, app.ID, testConfig.DefaultTenant)
+
+	t.Log("Prepare operation client for compass-system namespace")
+	cfg, err := rest.InClusterConfig()
+	require.NoError(t, err)
+	k8sClient, err := client.NewForConfig(cfg)
+	require.NoError(t, err)
+	operationsK8sClient := k8sClient.Operations("compass-system")
+
+	operationName := fmt.Sprintf("application-%s", app.ID)
+	t.Log(fmt.Sprintf("Check operation CR with name %s is created", operationName))
+	operation, err := operationsK8sClient.Get(ctx, operationName, metav1.GetOptions{})
+	require.NoError(t, err)
+	require.NotEmpty(t, operation)
+
+	t.Log(fmt.Sprintf("Verify operation CR with name %s is in progress", operationName))
+	counter := 0
+	isInDesiredState, webhookResponse = isWebhookOperationInDesiredState(t, operationFullPath, webhook.OperationResponseStatusINProgress)
+	for (!isInDesiredState) && counter < 100 {
```

Comment:
Instead of maintaining a counter and looping with `for` you can leverage `require.Eventually`, something we did in SM as well. Here's an example:

```
require.Eventually(t, func() bool {
		// logic here -> function will run every 5 seconds for 2minutes until the function returns `true`
	}, 2*time.Minute, 5*time.Second)
```

https://github.com/kyma-incubator/compass/blob/59756f9b7ca08a29d21f3de68ff2c9584f3284ea/components/operations-controller/controllers/operation_controller_integration_test.go#L157